docs/content/en/docs/yaml-crd-ref/task.md

```diff
@@ -0,0 +1,206 @@
+    appVersion: "1.0.0"
+    objectType: ""
+    taskType: ""
+    workloadName: "my-workload"
+    workloadVersion: "1.0.0"
+```
+
+## Fields
+
+* **apiVersion** -- API version being used.
+`
+* **kind** -- Resource type.
+   Must be set to `KeptnTask`
+
+* **metadata**
+  * **name** -- Unique name of this run of the task.
+    This name must be modified each time you run this `KeptnTask`,
+    so a common practice is to add a number to the end of the string
+    used for the `name` field of the `KeptnTaskDefinition` resource
+    so you can increment the number for each run.
+    Names must comply with the
+    [Kubernetes Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)
+    specification.
+* **spec** - Defines the desired state of this `KeptnTask` resource
+  * **workload** - Name of the
+      [KeptnWorkload](../crd-ref/lifecycle/v1alpha3/#keptnworkload)
+      resource for which the `KeptnTask` is being executed.
+
+      TODO: API ref shows `workloadName` and `workloadVersion`
+      in `TaskContext`, not directly in `spec`.
+      And I don't see a plain `workload` field.
+
+  * **workloadVersion** - `KeptnWorkload` version
+      for which the `KeptnTask` is being executed.
+  * **appName** - Name of the
+      [KeptnApp](../yaml-crd-ref/app.md) resource
+      for which the `KeptnTask` is being executed.
+  * **appVersion** - Version of the `KeptnApp` resource
+      for which the `KeptnTask` is being executed.
+
+      TODO: So I can create a `KeptnApp` resource that lists multiple workloads
+      to be processed?
+       Do I need to annotate the Workloads
+      and can I autocreate the `KeptnApp` or do I need to do it manually?
+
+      TODO: API reference shows `appName` and `appVersion` in `TaskContext`
+
+      TODO: API reference shows `parameters` and `secureParameters` here.
+      Should they be added to this list?
+
+      TODO: API reference also shows `CheckType`, `retries`, and `timeout` here.
+
+  * **taskDefinition** - Name of the corresponding `KeptnTaskDefinition` resource.
+  This `KeptnTaskDefinition` can be located in the same namespace
+  or in the Keptn name space.
+    * **context** - Contextual information about the task execution
+      * **appName** - Name of the
+          [KeptnApp](../yaml-crd-ref/app.md) resource
+          for which the `KeptnTask` is being executed.
+      * **appVersion** - Version of the `KeptnApp` resource
+          for which the `KeptnTask` is being executed.
+
+          TODO: So I can create a `KeptnApp` resource that lists multiple workloads
+          to be processed?
+           Do I need to annotate the Workloads
+          and can I autocreate the `KeptnApp` or do I need to do it manually?
+      * **objectType** - Indicates whether this `KeptnTask`
+          is being executed for a `KeptnApp` or a `KeptnWorkload` resource.
+
+          TODO: So why is this set to null (`""`) in the example?
```

Comment:
Because I can't find where the suitable types for this field are documented and an empty string just happens to work. This is one for whomever coded that piece.

Reply:
Okay.  I'm removing the TODO's from this page so we can get it merged.  We can improve it later.